crates/runix/src/flake_ref/mod.rs

```diff
@@ -286,5 +490,126 @@ pub(super) mod tests {
+    fn test_resolve_absolute_non_git_local() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let basic = flake_test_dir.path().canonicalize().unwrap().join("basic");
+        fs::create_dir_all(&basic).unwrap();
+        File::create(basic.join("flake.nix")).unwrap();
+
+        let basic = basic.to_string_lossy();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&basic).unwrap().to_string(),
+            Url::parse(&format!("path:{basic}")).unwrap().to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_relative_non_git_local() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let basic = flake_test_dir.path().canonicalize().unwrap().join("basic");
+        fs::create_dir_all(&basic).unwrap();
+        File::create(basic.join("flake.nix")).unwrap();
+
+        let path = pathdiff::diff_paths(&basic, env::current_dir().unwrap()).unwrap();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&path.to_string_lossy())
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("path:{}", basic.to_string_lossy()))
+                .unwrap()
+                .to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_absolute_git_local() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let git_dir = flake_test_dir
+            .path()
+            .canonicalize()
+            .unwrap()
+            .join("withgit");
+        fs::create_dir_all(&git_dir).unwrap();
+        fs::create_dir_all(git_dir.join(".git")).unwrap();
+        File::create(git_dir.join("flake.nix")).unwrap();
+
+        let git_dir_string = git_dir.to_string_lossy();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&git_dir_string)
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("git+file:{git_dir_string}"))
+                .unwrap()
+                .to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_absolute_git_local_nested() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let flake_dir_name = "inner";
+
+        let git_dir = flake_test_dir
+            .path()
+            .canonicalize()
+            .unwrap()
+            .join("withgit");
+        fs::create_dir_all(&git_dir).unwrap();
+        fs::create_dir_all(git_dir.join(".git")).unwrap();
+
+        let flake_root = git_dir.join(flake_dir_name);
+        fs::create_dir_all(&flake_root).unwrap();
+
+        File::create(flake_root.join("flake.nix")).unwrap();
+
+        let git_dir_string = git_dir.to_string_lossy();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&flake_root.to_string_lossy())
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("git+file:{git_dir_string}?dir={flake_dir_name}"))
+                .unwrap()
+                .to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_relative_git_local() {
```

Comment:
Combine this with `test_resolve_absolute_git_local`? The setup is the same and then you can assert that all three things are equal

Reply:
Not quite sure how to combine them without making a bit of an awkward setup: 

```
withgit
├── .git
├── flake.nix
└── inner
  └── flake.nix
```

I think i prefer testing those separately for now

Reply:
Not quite sure how to combine them without making a bit of an awkward setup: 

```
withgit
├── .git
├── flake.nix
└── inner
  └── flake.nix
```

I think i prefer testing those separately for now